### PR TITLE
Eliminating shellcheck failures

### PR DESCRIPTION
**Pull request by ds-ms**
[shellcheck](http://shellcheck.net/) is a linter for Bourne shell scripts that locates code that may be problematic under certain circumstances like accidental word splitting, globbing or exit code validation.

I ran shellcheck across all the `.sh` files in the repository, and made fixes to eliminate those failures. Please validate your scenarios. 

I am going to wire CI for this repo in `mseng`, this PR is necessary to ensure that the CI is green. 

See #46 for more details.

### CODE REVIEW

**Comment by vincentdass** [comment]
validated web app & container app.

File: aks/entrypoint.sh [hunk 1]
```diff
@@ -68,11 +68,15 @@ if [ -z "$HELM_CHART_PATH" ]; then
...
 
   if [ -n "${INGRESS_ROUTING_ZONE}" ]; then
-    DEFAULT_ARGS="--set image.repository=$CONTAINER_IMAGE_NAME --set ingress.enabled=true --set ingress.hostname=${HELM_RELEASE_NAME}.${INGRESS_ROUTING_ZONE}"
+    DEFAULT_ARGS[1]="ingress.enabled=true"
+    DEFAULT_ARGS[2]="ingress.hostname=${HELM_RELEASE_NAME}.${INGRESS_ROUTING_ZONE}"
   fi
-fi
 
-helm upgrade --install --force $DEFAULT_ARGS $* $HELM_RELEASE_NAME $HELM_CHART_PATH
+  helm upgrade --install --force --set "$(IFS=,; echo "${DEFAULT_ARGS[*]}")" "$@" $HELM_RELEASE_NAME $HELM_CHART_PATH
```
Comment (bansalaseem) [inline aks/entrypoint.sh]: test for multiple args

File: containerwebapp/entrypoint.sh [hunk 2]
```diff
@@ -43,28 +43,20 @@ fi
...
+    CONTAINER_IMAGE_NAME="${CONTAINER_IMAGE_NAME}:${CONTAINER_IMAGE_TAG}"
 fi
 
 echo "Initiating Container deployment..."
 
-az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" $AZCLI_ARGUMENT
+az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" \
+    --docker-custom-image-name "${CONTAINER_IMAGE_NAME}" \
+    --docker-registry-server-user "${DOCKER_USERNAME}" \
+    --docker-registry-server-password "${DOCKER_PASSWORD}" \
```
Comment (bansalaseem) [inline containerwebapp/entrypoint.sh]: handle empty values explicitly and dont rely on az handling as we dont control it.

File: functionscontainer/entrypoint.sh [hunk 3]
```diff
@@ -52,28 +52,20 @@ then
...
+    CONTAINER_IMAGE_NAME="${CONTAINER_IMAGE_NAME}:${CONTAINER_IMAGE_TAG}"
 fi
 
 echo "Initiating Container deployment..."
 
-az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" $AZCLI_ARGUMENT
+az webapp config container set -n "${AZURE_APP_NAME}" -g "${RESOURCE_GROUP_NAME}" \
+    --docker-custom-image-name "${CONTAINER_IMAGE_NAME}" \
+    --docker-registry-server-user "${DOCKER_USERNAME}" \
+    --docker-registry-server-password "${DOCKER_PASSWORD}" \
```
Comment (bansalaseem) [inline functionscontainer/entrypoint.sh]: same as earlier

File: containerwebapp/entrypoint.sh (hunk 2 above)
Comment (ds-ms) [reply containerwebapp/entrypoint.sh]: done

**Pull request merged**